airflow/www/security.py

```diff
@@ -71,6 +71,7 @@ class AirflowSecurityManager(SecurityManager, LoggingMixin):
         (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_DEPENDENCIES),
         (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_CODE),
         (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DATASET),
```

Comment:
Need to update `test_get_user_roles_for_anonymous_user`:

https://github.com/apache/airflow/blob/a5be64ad300ef99ab820ac020afba316f4c32613/tests/www/test_security.py?q=test_get_user_roles_for_anonymous_user#L375-L405